scanpy/external/tl/_execute_r_script.py

```diff
@@ -0,0 +1,121 @@
+from typing import Optional, Tuple
+from pathlib import Path
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import subprocess
+import os
+
+HERE = Path(__file__).parent
+
+
+def execute_r_script(
+    rscript_path: str = None,
+    r_filename: str = None,
+    arguments: Optional[list] = None,
+    verbosity: Optional[bool] = False,
+):
+    """\
+
+    Helper function that runs given R file in terminal using python subprocess module with the Rscript command
+
+    Parameters
+    ----------
+    rscript_path
+        Path to local directory of Rscript.exe
+    r_filename
+        Filename of R script to run
+    arguments
+        optional list of arguments to add to command
+    verbosity
+        Option to write subprocess output to terminal. Default false, which sets
+        stdout and stderr to NULL
+    cache @ TODO
+        Option to save intermediate R files. Default false.
+
+    Returns
+    -------
+    Boolean of whether script ran successfully
+
+    Notes
+    -------
+    This function requires the local path to the Rscript command to run the
+    provided script using the subprocess module
+
+    This function will prompt the user in terminal for the Rscript command path
+
+    Example
+    -------
+    >>> is_success = scanpy.external.tl.execute_r_script('PATH_TO_Rscript.exe', test.R')
+
+    """
+    ####  Just to print run time ####
+    import time
+    start_time = time.time()
+    ####  Just to print run time ####
+
+    if rscript_path is None:
+        raise ValueError('Please provide the local path to your Rscript.exe executable')
+    if r_filename is None:
+        raise ValueError('Please provide the filename of your R script')
+
+    try:
+
+        # create temp directory for r script files
+        temp_dir = Path(HERE, '_tmp')
+        if not os.path.exists(temp_dir):
+            os.mkdir(temp_dir)
+        print('temp_dir in python:', temp_dir)
+
+        path2rscript = Path(HERE, '_scripts', r_filename)
+
+        command = [rscript_path, path2rscript, '--vanilla', temp_dir]
+
+        # add optional arguments for the R script
+        if arguments is not None:
+            command += arguments
+
+        if not verbosity:
+            exit_code = subprocess.call(command, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
```

Comment:
I don't know how these work, but what happens if arguments that aren't valid get added to command? Is that possible? / Does it cause an error?